server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -450,10 +447,8 @@ public ActionModule(Settings settings, IndexNameExpressionResolver indexNameExpr
                 restWrapper = newRestWrapper;
                 if (restWrapper.getClass().getCanonicalName() == null ||
                     restWrapper.getClass().getCanonicalName().startsWith("org.elasticsearch") == false) {
-                    deprecationLogger.deprecate(DeprecationCategory.PLUGINS, "3rd_party_rest_deprecation", "The " +
-                        plugin.getClass().getName() + "plugin installs a custom REST wrapper. This functionality is deprecated and will " +
-                        "not be possible in Elasticsearch 8.0. If this plugin is intended to provide security features for Elasticsearch " +
-                        "then you should switch to using the built-in Elasticsearch features instead.");
+                    logger.debug("The " + plugin.getClass().getName() + "plugin installs a custom REST wrapper. This functionality " +
+                        "should be disabled");
```

Comment:
Having a debug here seems weird to me. I feel like we should remove the check entirely, or make in a warning/error.
Maintaining this code, just to print a debug message that no one will see doesn't seem valuable.